api/src/pcapi/core/educational/models.py

```diff
@@ -1380,8 +1380,23 @@ class CollectiveStock(PcObject, Base, Model):
 
     priceDetail = sa.Column(sa.Text, nullable=True)
 
+    @property
+    def lastBooking(self) -> "CollectiveBooking | None":
+        bookings = sorted(self.collectiveBookings, key=lambda booking: booking.dateCreated)
+        return bookings[-1] if bookings else None
+
+    @property
+    def lastBookingStatus(self) -> CollectiveBookingStatus | None:
+        booking = self.lastBooking
+        return booking.status if booking else None
+
     @property
     def isBookable(self) -> bool:
+        if (
+            FeatureToggle.ENABLE_COLLECTIVE_NEW_STATUSES.is_active()
+            and self.lastBookingStatus == CollectiveBookingStatus.CANCELLED
+        ):
+            return False
```

Comment:
J'ai un doute, ce n'est pas déjà couvert par `not self.isSoldOut` ?

Reply:
isSoldout => (isBookable = False) uniquement avec le FF

On pourrait changer la condition `self.lastBookingStatus == CollectiveBookingStatus.CANCELLED` en isSoldOut mais je ne trouve pas qu'on y gagne en lisibiité

Reply:
En effet si on a uniquement des bookings CANCELLED -> isSoldOut est False et isBookable est True en l'état
Avec ton changement et FF ON -> isBookable est False